proposed/http-client/http-client.md

```diff
@@ -21,9 +21,9 @@ implementations. This would make libraries more stable since the reduced number
 dependencies and the likelihood to get in version conflicts is reduced.
 
 The second goal is that all HTTP clients should follow the [Liskov substitutions principle][Liskov].
-This means that all clients should act the same when sending a request. By default a HTTP client
-should not follow redirect nor throw exceptions on HTTP responses with status 4xx or 5xx.
-
+This means that all clients should act the same when sending a request. A consuming
+library SHOULD expect a HTTP client to NOT follow redirects NOR throw exceptions based
+on HTTP response statuses.
```

Comment:
Good that you mention 1XX responses, because they're the ones that definitely need special treatment, see https://tools.ietf.org/html/rfc7231#section-6.2.